java/core/src/main/java/com/whylogs/core/metrics/Metric.java

```diff
@@ -0,0 +1,48 @@
+package com.whylogs.core.metrics;
+
+import com.whylogs.core.PreProcessedColumn;
+import com.whylogs.core.SummaryConfig;
+
+import java.util.HashMap;
+
+public abstract class Metric{
+
+    private String namespace;
+
+    // TODO: needs a __post_init__ for the registry
```

Comment:
Maybe not--the `__post_init__()` was a bit of a kludge to cope with interaction between `@property` and `@dataclass`. It would be worth trying to do the registration in the base class ctor(s) in the Java version.